File: leetcode/程序员面试经典/面试题 10.01. 合并排序的数组/2.java
class Solution {
    public void merge(int[] A, int m, int[] B, int n) {
        int p1 = m-1;
        int p2 = n-1;
        int p = m+n-1;
        while(p1>=0&&p2>=0){
            if(A[p1]>B[p2]){
                A[p--] = A[p1--];
            }else{
                A[p--]= B[p2--];
            }
        }
        int i = 0;
        while(i<=p2){
            A[i] = B[i];
            i++;
        }
        
    }
}

File: leetcode/剑指offer/剑指 Offer 05. 替换空格/2.cpp
class Solution {
public:
    string replaceSpace(string s) {
        string res;
        for(int i =0;i<s.size();i++){
            if(s[i]!=' '){
                res+=s[i];
            }else{
                res+="%20";
            }
        }
        return res;
    }
};